notify/receivers.go

```diff
@@ -292,3 +316,222 @@ func ProcessNotifierError(config *GrafanaReceiver, err error) error {
 
 	return err
 }
+
+// GrafanaReceiverTyped represents a parsed and validated APIReceiver
+type GrafanaReceiverTyped struct {
+	Name                string
+	AlertmanagerConfigs []*NotifierConfig[alertmanager.Config]
+	DingdingConfigs     []*NotifierConfig[dinding.Config]
+	DiscordConfigs      []*NotifierConfig[discord.Config]
+	EmailConfigs        []*NotifierConfig[email.Config]
+	GooglechatConfigs   []*NotifierConfig[googlechat.Config]
+	KafkaConfigs        []*NotifierConfig[kafka.Config]
+	LineConfigs         []*NotifierConfig[line.Config]
+	OpsgenieConfigs     []*NotifierConfig[opsgenie.Config]
+	PagerdutyConfigs    []*NotifierConfig[pagerduty.Config]
+	PushoverConfigs     []*NotifierConfig[pushover.Config]
+	SensugoConfigs      []*NotifierConfig[sensugo.Config]
+	SlackConfigs        []*NotifierConfig[slack.Config]
+	TeamsConfigs        []*NotifierConfig[teams.Config]
+	TelegramConfigs     []*NotifierConfig[telegram.Config]
+	ThreemaConfigs      []*NotifierConfig[threema.Config]
+	VictoropsConfigs    []*NotifierConfig[victorops.Config]
+	WebhookConfigs      []*NotifierConfig[webhook.Config]
+	WecomConfigs        []*NotifierConfig[wecom.Config]
+	WebexConfigs        []*NotifierConfig[webex.Config]
+}
+
+// NotifierConfig represents parsed GrafanaReceiver.
+type NotifierConfig[T interface{}] struct {
+	receivers.Metadata
+	Settings T
+}
+
+// ValidateAPIReceiver parses, decrypts and validates the APIReceiver. GrafanaReceiverTyped that contains configurations of all notifiers configurations for this receiver
+func ValidateAPIReceiver(ctx context.Context, api *APIReceiver, decrypt receivers.GetDecryptedValueFn) (GrafanaReceiverTyped, error) {
```

Comment:
This function is tricky to test thoroughly because it will mostly test NewConfig. Although I generally cover everything possible with tests, in this case I decided that it is not worth it because of the reason above. Anyway, if you think this does make sense, I will add tests.